core/workflow-core/src/main/scala/edu/uci/ics/amber/core/storage/result/iceberg/IcebergDocument.scala

```diff
@@ -26,7 +26,7 @@ import scala.jdk.CollectionConverters._
   * @param deserde function to deserialize an Iceberg Record into T.
   * @tparam T type of the data items stored in the Iceberg table.
   */
-private[storage] class IcebergDocument[T >: Null <: AnyRef](
```

Comment:
Why this change is needed? We intentionally make this document as private, because we want all the creation of the document to go through `DocumentFactory.createDocument`.